redbot/cogs/warnings/warnings.py

```diff
@@ -386,6 +387,34 @@ async def warn(
         or a custom reason if ``[p]warningset allowcustomreasons`` is set.
         """
         guild = ctx.guild
+        member = None
+        if isinstance(user, discord.Member):
+            member = user
+        elif isinstance(user, int):
+            user_obj = self.bot.get_user(user) or discord.Object(id=user)
+            try:
+                confirm = ConfirmView(ctx.author, timeout=10)
+                confirm.message = await ctx.send(
+                    f"User `{user}` is not in the server but has been found globally. Would you like to ban them instead?",
+                    view=confirm,
+                )
+                await confirm.wait()
+                if confirm.result is None:
+                    await ctx.send("No response received within 10 seconds. No action taken.")
+                    return
+                elif confirm.result:
+                    await ctx.guild.ban(user_obj, reason=reason)
+                else:
+                    confirm.message = await ctx.send("No action taken.")
+
+                await confirm.message.add_reaction("\N{WHITE HEAVY CHECK MARK}")
```

Comment:
If the bot does not have `add_reactions` permissions, this will cause it to claim an error occurred while trying to ban a user, which isn't quite true. Either this should use `await ctx.tick()` which handles that case correctly, or it should send a message instead to confirm.

Reply:
yep, now I added a checker